demo/Demo.cpp

```diff
@@ -0,0 +1,185 @@
+        nullptr,
+        std::make_unique<render::primitive::LineSegment>(
+            glm::translate(glm::mat4(1), glm::vec3(particle.GetPosition())), glm::vec3(0.439, 0.502, 0.565), start, end)
+    );
+
+    return m_objects.back();
+}
+
+Demo::Object& Demo::MakePlane(Particle particle, glm::dvec3 normal)
+{
+    m_objects.emplace_back(
+        &MakeRigidBody(particle, std::make_unique<geometry::Plane>(particle.GetPosition(), normal)),
+        std::make_unique<render::primitive::Plane>(
+            glm::translate(glm::mat4(1), glm::vec3(particle.GetPosition())), glm::vec3(0.439, 0.502, 0.565), normal)
+    );
+    m_particleContactGenerators.emplace_back(
+        std::make_unique<ShapeContactGenerator<RigidBodies>>(*m_objects.back().body, m_rigidBodies, 0.7)
+    );
+
+    return m_objects.back();
+}
+
+Demo::Object& Demo::MakeSphere(Particle particle, double radius)
+{
+    m_objects.emplace_back(
+        &MakeRigidBody(particle, std::make_unique<geometry::Sphere>(particle.GetPosition(), radius)),
+        std::make_unique<render::primitive::Sphere>(
+            glm::translate(glm::mat4(1), glm::vec3(particle.GetPosition())), glm::vec3(0.439, 0.502, 0.565), radius)
+    );
+    m_particleContactGenerators.emplace_back(
+        std::make_unique<ShapeContactGenerator<RigidBodies>>(*m_objects.back().body, m_rigidBodies, 0.7)
+    );
+
+    return m_objects.back();
+}
+
+Demo::Object& Demo::MakeBox(Particle particle, glm::vec3 i, glm::vec3 j, glm::vec3 k)
+{
+    m_objects.emplace_back(
+        &MakeRigidBody(particle, std::make_unique<geometry::Box>(particle.GetPosition(), i, j, k)),
+        std::make_unique<render::primitive::Box>(
+            glm::translate(glm::mat4(1), glm::vec3(particle.GetPosition())),
+            glm::vec3(0.439, 0.502, 0.565),
+            render::primitive::Box::Axes{i, j, k}
+        )
+    );
+    m_particleContactGenerators.emplace_back(
+        std::make_unique<ShapeContactGenerator<RigidBodies>>(*m_objects.back().body, m_rigidBodies, 0.7)
+    );
+
+    return m_objects.back();
+}
+
+void Demo::Remove(Object& object)
+{
+    if (object.body != nullptr)
+    {
+        Particle& particle = object.body->p;
+        m_particleForceRegistry.Remove(particle);
+
+        for (auto it = m_rigidBodies.begin(); it != m_rigidBodies.end(); ++it)
+        {
+            if (&it->p == &particle)
+            {
+                m_rigidBodies.erase(it);
+                break;
+            }
+        }
+        for (auto it = m_particles.begin(); it != m_particles.end(); ++it)
+        {
+            if (&*it == &particle)
+            {
+                m_particles.erase(it);
+                break;
+            }
+        }
+    }
+
+    for (auto it = m_objects.begin(); it != m_objects.end(); ++it)
+    {
+        if (&*it == &object)
+        {
+            m_objects.erase(it);
+            break;
+        }
+    }
+
+    m_particleContactGenerators.clear();
+    for (Object& obj : m_objects)
```

Comment:
Do you really need to reinitialize the whole thing for each deleted object? Am I missing something or you can do `std::find_if` and check the `unique_ptr`?